src/glmfit.jl

```diff
@@ -820,3 +820,38 @@ function checky(y, d::Binomial)
     end
     return nothing
 end
+
+const _RESIDUAL_TYPES = [:deviance, :pearson, :response, :working]
+
+"""
+    residuals(model::GeneralizedLinearModel; type=:deviance)
+
+Return the residuals of a GLM.
+
+Supported values for `type` are:
+- `:deviance` (the default): the signed square root of the element-wise
+  contribution to the deviance
+- `:response`: the difference between the observed and fitted values
+- `:working`: working residuals (used during the IRLS process)
+- `:pearson`: Pearson residuals, i.e., response residuals scaled
+    by the standard error of the response
+"""
+function residuals(model::GeneralizedLinearModel; type=:deviance)
+    type in _RESIDUAL_TYPES ||
```

Comment:
Why not use the `else` in the conditional to catch unsupported types rather than maintaining a separate list of types? Users can consult the docstring if they're not sure what's supported. The only other place the constant is used is in the tests where you loop over it. You could just move the list there where it's needed.